src/register/Register.jsx

```diff
@@ -1,291 +1,331 @@
 import React, { useState, useContext } from "react";
 import "./Register.css";
+import Footer from "../home/Footer";
 import { Redirect, Link } from "react-router-dom";
 import { AuthContext } from "../AuthContext";
-import {BASE_API} from "../config";
-import {SERVICE_UNAVAILABLE_ERROR} from "../messages";
+import { BASE_API } from "../config";
+import { SERVICE_UNAVAILABLE_ERROR } from "../messages";
 import TermsAndPrivacyPolicyModal from "./TermsAndPrivacyPolicyModal";
+import hidePwd from "../assets/images/Hide_pwd_icon.svg";
+import showPwd from "../assets/images/show_pwd_icon.svg";
 
 export default function Register() {
-    const {isAuth} = useContext(AuthContext);
-    const [isValidName, setIsValidName] = useState(true);
-    const [isValidUsername, setIsValidUsername] = useState(true);
-    const [isValidEmail, setIsValidEmail] = useState(true);
-    const [isValidPassword, setIsValidPassword] = useState(true);
-    const [isValidConfirmPassword, setIsValidConfirmPassword] = useState(true);
-    const [responseMessage, setResponseMessage] = useState(null);
-    const [password, setPassword] = useState("");
-    const [confirmPassword, setConfirmPassword] = useState("");
-    const [isPasswordShown, setIsPasswordShown] = useState(false);
-    const [showTermsModal,setShowTermsModal] = useState(false);
+  const { isAuth } = useContext(AuthContext);
+  const [isValidUsername, setIsValidUsername] = useState(true);
+  const [isValidName, setIsValidName] = useState(true);
```

Comment:
if these are validators, does it make more sense for initial state to be set to `false` instead? they should only be `true` after successful validation

Reply:
@rkraeher Thank you for reviewing. Values are set so when the user enters the wrong value then display an error. Apart from validating, error message is also handled by this.